sample-android-app/src/main/java/com/badoo/reaktive/sample/android/MainActivity.kt

```diff
@@ -12,12 +21,41 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        ObservablePlugins.onAssemble =
```

Comment:
My idea was to keep the API as simple as possible. With new "single" approach one still can create custom list of plugins as follows:
```
object MySinglePlugins : OnAssembleSingle {
    val onAssemble = ArrayList<OnAssembleSingle>()

    override fun <T> wrap(single: Single<T>): Single<T> =
        onAssemble.fold(single) { src, plugin -> plugin.wrap(src) }
}

fun onCreate() {
    SinglePlugins.onAssemble = MySinglePlugins
}
```

RxJava also provides similar API. But there is a drawback: it is not possible to have independent plugin libraries, so each adds its own plugins to the global list. Like with `Traceur`, it provides `Traceur.enableLogging()` method, which registers its internals via e.g. `RxJavaPlugins.setOnObservableAssembly`. Looks like you can use another similar "plugin" library, they will overwrite each other.

What do you think?

Reply:
Can plugins decorate/compose each other?

Reply:
Yeah, looks like they can! I will think about an extension for this. Thanks!

Reply:
@mk-jones 
1. Yes, something like this. Except ideally we should allow third-parties to add their plugins, instead of aggregating them on client side. To prevent cases like with Traceur, it just replaces any currently assigned plugin.
2. This was done intentionally, as I wanted to separate concerns. But I will think more about it. Thanks!

Reply:
@CherryPerry @mk-jones @saket Addressed all the comments. Could you please check again?